akka-sample-fsm-scala/src/main/scala/sample/DiningHakkers.scala

```diff
@@ -54,57 +62,57 @@ object Chopstick {
   }
 }
 
-/**
-  * Some fsm hakker messages
-  */
-sealed trait HakkerMessage
-case object Think extends HakkerMessage
-case object Eat extends HakkerMessage
+object Hakker {
+  sealed trait HakkerMessage
+  case object Think extends HakkerMessage
+  case object Eat extends HakkerMessage
+  final case class ChopstickAnswerAdaptor(msg: ChopstickAnswer) extends HakkerMessage
```

Comment:
Better classname would be `AdaptedAnswer(answer)` (thinking of it as an event) or `HandleAnswer(answer)` (thinking of it as command, then all three messages would be commands and top type could be `Command`)